app/components-react/windows/Projector.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react';
+import electron from 'electron';
+import * as remote from '@electron/remote';
+import { ModalLayout } from 'components-react/shared/ModalLayout';
+import Display from 'components-react/shared/Display';
+import Util from 'services/utils';
+import Scrollable from 'components-react/shared/Scrollable';
+import { Services } from 'components-react/service-provider';
+import { useVuex } from 'components-react/hooks';
+import styles from './Projector.m.less';
+
+export default function Projector() {
+  const { WindowsService, SourcesService } = Services;
+
+  const [oldBounds, setOldBounds] = useState<electron.Rectangle | null>(null);
+
+  const windowId = Util.getCurrentUrlParams().windowId;
+  const { hideStyleBlockers, fullscreen, sourceId, renderingMode } = useVuex(() => {
+    return {
+      hideStyleBlockers: WindowsService.state[windowId].hideStyleBlockers,
+      fullscreen: WindowsService.state[windowId].isFullScreen,
+      sourceId: WindowsService.getWindowOptions(windowId).sourceId,
+      renderingMode: WindowsService.getWindowOptions(windowId).renderingMode,
+    };
+  });
+
+  useEffect(() => {
+    const sourcesSub = SourcesService.sourceRemoved.subscribe(source => {
+      if (source.sourceId === sourceId) {
+        remote.getCurrentWindow().close();
+      }
+    });
+
+    return sourcesSub.unsubscribe;
+  }, []);
```

Comment:
Could use `useSubscription` here.  One thing we should probably get better at is using higher level general hooks when possible.  Improves readability I think.